punishments.js

```diff
@@ -1098,13 +1098,16 @@ Punishments.checkName = function (user, registered) {
  * @param {Connection} connection
  */
 Punishments.checkIp = function (user, connection) {
+	// If connection.user is reassigned before this can finish running, user
+	// will no longer be equal to it.
+	user = connection.user || user;
```

Comment:
There's no reason to do this at the top...... in what case would the user be reassigned here?

Reply:
Probably none